test/dx_mgn_pool.js

```diff
@@ -296,5 +311,41 @@ contract("DxMgnPool", (accounts) => {
 
       await truffleAssert.reverts(instance.withdrawMagnolia(), "MGN has not been unlocked, yet")
     })
+    it("withdraws the right amount of MGN", async () => {
+      const depositTokenMock = await MockContract.new()
+      const secondaryTokenMock = await MockContract.new()
+      const mgnTokenMock = await MockContract.new()
+      const mgnToken = await TokenFRT.deployed()
+      const dxMock = await MockContract.new()
+      const dx = await DutchExchange.new()
+      const poolingEndBlock = (await web3.eth.getBlockNumber()) + 100
+      const instance = await DxMgnPool.new(depositTokenMock.address, secondaryTokenMock.address, mgnTokenMock.address, dxMock.address, poolingEndBlock)
+      
+      await depositTokenMock.givenAnyReturnBool(true)
```

Comment:
I think a clean structure for a unit test is:
1) instantiation of component/dependent mocks
2) setting up the mocks with `given...` (grouped by mock)
3) interactions
4) assertions/expectation checking